### PR TITLE
DOC Add some extra details to the changelog about toast notification and DataObject hydration

### PR DESCRIPTION
**Pull request by maxime-rainville**

### CODE REVIEW

File: docs/en/04_Changelogs/4.7.0.md [hunk 1]
```diff
@@ -76,3 +76,35 @@ An edgecase exists where a project can update to `silverstripe/framework 4.7.0`
...
+- actions can now be attached to the toast notifications
+- toast notification are now accessible for all users.
+
+Asset-admin has been updated to use toast notifications instead of alerts. More modules will start using them in future releases.
+
+[Review the Toast story in the pattern library](https://silverstripe.github.io/silverstripe-admin/?selectedKind=Admin%2FToasts&selectedStory=Single) to learn how and when to use the toast notifications.
+
+### DataObject constructor support for hydration
+
+Many developers wrongly assumed that the DataObject constructor could be used to "hydrate" a record by passing an array of value to it. While this worked in somecases, values would be run throught setters and validation occasinally leading to inconsitencies.
```
Comment (brynwhyman) [inline docs/en/04_Changelogs/4.7.0.md:93]:
```suggestion
Many developers wrongly assumed that the DataObject constructor could be used to "hydrate" a record by passing an array of value to it. While this worked in somecases, values would be run through setters and validation occasinally leading to inconsitencies.
```

File: docs/en/04_Changelogs/beta/4.7.0-beta1.md [hunk 2]
```diff
@@ -77,6 +77,38 @@ An edgecase exists where a project can update to `silverstripe/framework 4.7.0`
...
+- actions can now be attached to the toast notifications
+- toast notification are now accessible for all users.
+
+Asset-admin has been updated to use toast notifications instead of alerts. More modules will start using them in future releases.
+
+[Review the Toast story in the pattern library](https://silverstripe.github.io/silverstripe-admin/?selectedKind=Admin%2FToasts&selectedStory=Single) to learn how and when to use the toast notifications.
+
+### DataObject constructor support for hydration
+
+Many developers wrongly assumed that the DataObject constructor could be used to "hydrate" a record by passing an array of value to it. While this worked in somecases, values would be run throught setters and validation occasinally leading to inconsitencies.
```
Comment (brynwhyman) [inline docs/en/04_Changelogs/beta/4.7.0-beta1.md:93]:
```suggestion
Many developers wrongly assumed that the DataObject constructor could be used to "hydrate" a record by passing an array of value to it. While this worked in some cases, values would be run through setters and validation occasionally leading to inconsistencies.
```

**Review by brynwhyman — CHANGES_REQUESTED** [review]
Minor spelling issues, otherwise LGTM

**Review by Cheddam — CHANGES_REQUESTED** [review]
Please also format to max 120 character lines :)

File: docs/en/04_Changelogs/4.7.0.md [hunk 3]
```diff
@@ -76,3 +76,35 @@ An edgecase exists where a project can update to `silverstripe/framework 4.7.0`
...
+
+### DataObject constructor support for hydration
+
+Many developers wrongly assumed that the DataObject constructor could be used to "hydrate" a record by passing an array of value to it. While this worked in some cases, values would be run through setters and validation occasionally leading to inconsistencies.
+
+Silverstripe CMS Recipe 4.7 introduces improvements to the DataObject constructor to provide proper support for hydration.
+
+```php
+// Default behavior. Setters and validation will be applied to the provided data.
+// Equivalent to `Foo::create(['Title' => 'Bar'])` or `new Foo(['Title' => 'Bar'])`
```
Comment (michalkleiner) [inline docs/en/04_Changelogs/4.7.0.md:99]: I'd probably remove the `new Foo` example and only say it's equivalent to the `::create` example, or mention possible Injector related differences. My understanding was that using Injector was the preferred way anyway, so the least examples of the `new Foo` approach the better.

**Comment by Cheddam** [comment]
@maxime-rainville One more thing - these sections need to be added to the table of contents at the top of the doc.

File: docs/en/04_Changelogs/4.7.0.md (hunk 3 above)
Comment (maxime-rainville) [reply docs/en/04_Changelogs/4.7.0.md]: Fixed

**Review by brynwhyman — APPROVED** [review]

**Review by Cheddam — APPROVED** [review]

**Pull request merged**